### PR TITLE
Build extension command tree

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Add scripts to build a command tree json for all extensions and upload to a storage account.

These scripts will be used in the release sync pipeline of extensions. Test pipeline: https://dev.azure.com/azure-sdk/internal/_build?definitionId=1857&_a=summary

The json file will be used in Azure CLI, related PR: https://github.com/Azure/azure-cli/pull/14478

An example of the built file:
```
{
       "aks": {
               "create": "aks-preview",
               "update": "aks-preview",
               "app": {
                    "up": "deploy-to-azure"
               },
               "use-dev-spaces": "dev-spaces"
        },
        "blueprint": {
               "create":"blueprint",
               "update":"blueprint"
         },
        ...
}
```
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by haroldrandom** [comment]
Who's going to need this?

**Comment by jiasli** [comment]
I think the naming may cause confusion with the existing **command index** (https://github.com/Azure/azure-cli/pull/13294).

**Comment by fengzhou-msft** [comment]
> I think the naming may cause confusion with the existing **command index** ([Azure/azure-cli#13294](https://github.com/Azure/azure-cli/pull/13294)).

How about the name **command tree** as it mimics a [prefix tree](https://en.wikipedia.org/wiki/Trie)?

**Comment by fengzhou-msft** [comment]
> Who's going to need this?

PR:https://github.com/Azure/azure-cli/pull/14478 needs to search command in a json file for all extension commands and the json file is built in this PR.

**Comment by yonzhan** [comment]
Trie tree solution is good and the performance is fast.

File: scripts/ci/update_ext_cmd_tree.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                                 file_path=file_path)
+
+    url = client.make_blob_url(container_name=STORAGE_CONTAINER, blob_name=blob_file_name)
+
+    download_file_path = os.path.expanduser(os.path.join('~', '.azure', downloaded_file_name))
+    download_file(url, download_file_path)
+    if filecmp.cmp(file_path, download_file_path):
+        print("extensionCommandTree.json uploaded successfully. URL: {}".format(url))
+    else:
+        raise Exception("Failed to update extensionCommandTree.json in storage account")
```
Comment (fengzhou-msft) [inline scripts/ci/update_ext_cmd_tree.py:85]:
```suggestion
        raise Exception("Failed to update extensionCommandTree.json in the storage account")
```

**Pull request merged**